### PR TITLE
Fix #234 auto_session_create_enabled take a func

### LINKED ISSUE #234: [FEATURE] `auto_session_create_enabled` should also accept a function
**Is your feature request related to a problem? Please describe.**
Currently, specifying each directory in `auto_session_suppress_dirs` or `auto_session_allowed_dirs` is the only way to toggle auto session per directory.

**Describe the solution you'd like**
In addition to boolean values, `auto_session_create_enabled` should also accept a function that returns a boolean.  
This way, users will be able to put more logics about when to enable auto session.

**Example use case**
```lua
require("auto-session").setup({
  auto_session_create_enabled = function()
    local cmd = "git rev-parse --is-inside-work-tree"
    return vim.fn.system(cmd) == "true\n"
  end,
})
```
If said feature is implemented, this configuration will auto create sessions only when we're inside a Git repository.  
Personally, most of my sessions correspond to a project, and most of my projects have been `$ git init`ed.  
So it will automatically create sessions exactly when I want them to be created.

### PR DESCRIPTION
**Pull request by cameronr**
Let auto_session_create_enabled take a function that lets a user customize when a session can be created automatically

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1,4 +1,4 @@
-# 🗒️ Description
+# 🗒️ Descriptionread
```
Comment (rmagatti) [inline README.md:1]:
```suggestion
# 🗒️ Description
```

**Review by rmagatti — CHANGES_REQUESTED** [review]

File: README.md (hunk 1 above)
Comment (cameronr) [reply README.md:1]: Whoops, good catch!

**Review by rmagatti — APPROVED** [review]
💪

**Comment by cameronr** [comment]
Interesting, I guess fixing the README conflict locally (rather than on GH) dismissed your review. sorry about that.

**Review by rmagatti — APPROVED** [review]
No worries! Thanks!

**Pull request merged**